### PR TITLE
lint: do not install golangci-lint from source

### PR DESCRIPTION
**Pull request by xcoulon**
check if the `golangci-lint` executable is in the path
if not, fail with a error message including a link
to the install instructions.

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

**Comment by xcoulon** [comment]
@MatousJobanek @alexeykazakov once we agreed on the way to install/run `golangci-lint` locally, I will apply the same settings on other repos.

**Comment by xcoulon** [comment]
@alexeykazakov @MatousJobanek also, should we consider replacing the `ci/prow/lint` job with a GitHub Action, as we do on sandbox-sre? I believe the integration with GitHub will be better and we won't have to deal with installing `golangci-lint` when running the job on `ci/prow`

**Review by alexeykazakov — APPROVED** [review]

**Comment by alexeykazakov** [comment]
Yes let's move it to GH actions

File: make/lint.mk [hunk 1]
```diff
@@ -11,5 +11,7 @@ lint-yaml: ${YAML_FILES}
 .PHONY: lint-go-code
 ## Checks the code with golangci-lint
 lint-go-code: generate
-	$(Q)go get github.com/golangci/golangci-lint/cmd/golangci-lint
-	$(Q)${GOPATH}/bin/golangci-lint ${V_FLAG} run
\ No newline at end of file
+ ifeq (, $(shell which golangci-lint))
```
Comment (MatousJobanek) [inline make/lint.mk:14]: This will make the output more clear
```suggestion
 ifeq (, $(shell which golangci-lint 2>/dev/null))
```

**Review by MatousJobanek — APPROVED** [review]
Nice :+1:

**Comment by xcoulon** [comment]
> Yes let's move it to GH actions

see https://github.com/openshift/release/pull/18329 to remove the `ci/prow/lint` job

File: make/lint.mk (hunk 1 above)
Comment (xcoulon) [reply make/lint.mk]: thanks! done in e9c760c

**Pull request merged**